### PR TITLE
Mm feature/migrate content library item info

### PR DESCRIPTION
**Pull request by mikemorency**
##### SUMMARY
This is migrating the content_library_item_info from the vmware_rest collection to here. The auto-generated module in the rest collection does not work due to an issue with the OpenAPI spec, so its easier to add the functionality to this collection instead.

I also consolidated some of the content library shared methods in the rest api module_util

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
content_library_item_info

### CODE REVIEW

File: tests/integration/targets/vmware_content_library_item_info/vars.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+vcenter_hostname: "127.0.0.1"
```
Comment (anna-savina) [inline tests/integration/targets/vmware_content_library_item_info/vars.yml:1]: I don't see test for simulator. Will you add it later?

File: tests/integration/targets/vmware_content_library_item_info/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+          - (__res.library_item_info | length) > 0
+
+    - name: Gather Fake Library Info
+      vmware.vmware.content_library_item_info:
+        validate_certs: false
+        hostname: "{{ vcenter_hostname }}"
+        username: "{{ vcenter_username }}"
+        password: "{{ vcenter_password }}"
+        port: "{{ vcenter_port }}"
+        library_item_name: should_not_exist
```
Comment (anna-savina) [inline tests/integration/targets/vmware_content_library_item_info/tasks/main.yml:32]: Should be `library_name` specified?

File: tests/integration/targets/vmware_content_library_item_info/vars.yml (hunk 1 above)
Comment (mikemorency) [reply tests/integration/targets/vmware_content_library_item_info/vars.yml:1]: I cannot figure out how to mock the rest api for this module, so I dont think I can include the simulator tests at this time

Comment (anna-savina) [reply tests/integration/targets/vmware_content_library_item_info/vars.yml:1]: In this case better delete `vars.yml` and remove it from `run.yml`

**Review by anna-savina — APPROVED** [review]
LGTM

**Review by anna-savina — APPROVED** [review]

File: plugins/module_utils/_vmware_rest_client.py [hunk 3]
```diff
@@ -191,51 +193,54 @@ def get_vm_by_name(self, name):
 
         return vms[0]
 
-    def get_library_item_by_name(self, name):
+    def get_content_library_ids(self, name=None, library_type=None, fail_on_missing=False):
```
Comment (bardielle) [inline plugins/module_utils/_vmware_rest_client.py:196]: Why did you change this behavior?

Comment (bardielle) [reply plugins/module_utils/_vmware_rest_client.py:196]: I saw that the other functions are still by name
@machacekondra what do you think?

Comment (mariolenz) [reply plugins/module_utils/_vmware_rest_client.py:196]: @machacekondra @mikemorency I don't think so because they're all marked as private. If people use them and they break backward compatibility, that's their problem. That's what private means.

Apropos of nothing, it would be great if there would be some non-private module utils that I could re-use in `community.vmware`. Actually, I think this was the original plan :thinking:

Comment (mikemorency) [reply plugins/module_utils/_vmware_rest_client.py:196]: Yes we did make them private while we were still figuring out method naming and making changes. Sounds like we can stick with whats in the PR.

@mariolenz , Most of the pyvmomi/rest api base utils stay the same, but I still make the occasional functional change to the other utils. Ill make a ticket to review the function names and prep the pyvmomi/rest utils to make them public. Ill have to look at previous conversation to see if that can be done in a minor release or needs a major one
